File: QuestionAnswering/QA_ICL/frontend/next_app_marie/src/lib/model/comp-op.ts
export const OComparisonOperator = {
  EQ: 'eq',
  LT: 'lt',
  GT: 'gt',
  LTE: 'lte',
  GTE: 'gte',
}
export type ComparisonOperator =
  (typeof OComparisonOperator)[keyof typeof OComparisonOperator]

File: QuestionAnswering/QA_ICL/backend/fastapi_app/model/web/comp_op.py
from enum import Enum


class ComparisonOperator(str, Enum):
    EQ = "eq"
    LT = "lt"
    GT = "gt"
    LTE = "lte"
    GTE = "gte"


COMP_OP_2_SPARQL_SYMBOL = {
    ComparisonOperator.EQ: "=",
    ComparisonOperator.LT: "<",
    ComparisonOperator.GT: ">",
    ComparisonOperator.LTE: "<=",
    ComparisonOperator.GTE: ">=",
}
